### PR TITLE
simplify DefaultDiagnosticBuilder

### PR DESCRIPTION
**Pull request by gracew**

### CODE REVIEW

File: groovy-language-server/src/test/java/com/palantir/ls/util/DefaultDiagnosticBuilderTest.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
```
Comment (gracew) [inline groovy-language-server/src/test/java/com/palantir/ls/util/DefaultDiagnosticBuilderTest.java:2]: this needs to be changed to the apache license

File: groovy-language-server/src/main/java/com/palantir/ls/util/DefaultDiagnosticBuilder.java [hunk 2]
```diff
@@ -16,53 +16,13 @@
...
+public final class DefaultDiagnosticBuilder extends DiagnosticBuilder {
 
-    private final String message;
-    private final DiagnosticSeverity severity;
-
-    private Optional<String> code = Optional.absent();
-    private Optional<Range> range = Optional.absent();
-    private Optional<String> source = Optional.absent();
-
-    public DefaultDiagnosticBuilder(String message, DiagnosticSeverity warning) {
```
Comment (natgabb) [inline groovy-language-server/src/main/java/com/palantir/ls/util/DefaultDiagnosticBuilder.java:35]: The reason for the constructor with a message and warning is to disallow creation of diagnostics without the fields required by the protocol. All the other fields have defaults, or in the case of `code` is optional.

File: groovy-language-server/src/test/java/com/palantir/ls/util/DefaultDiagnosticBuilderTest.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
+/*
```
Comment (natgabb) [inline groovy-language-server/src/test/java/com/palantir/ls/util/DefaultDiagnosticBuilderTest.java:1]: Put Apache license

File: groovy-language-server/src/main/java/com/palantir/ls/util/DefaultDiagnosticBuilder.java (hunk 2 above)
Comment (gracew) [inline groovy-language-server/src/main/java/com/palantir/ls/util/DefaultDiagnosticBuilder.java:35]: ok, I'll add message and warning back to the constructor.

**Comment by gracew** [comment]
@natgabb updated PR

**Comment by jiahuijiang** [comment]
+1

**Comment by natgabb** [comment]
Nice +1

**Pull request merged**